From e9ba0c54f2c01bb530370c16502cf304fc18b3f6 Mon Sep 17 00:00:00 2001
From: Alexander Lohnau <alexander.lohnau@gmx.de>
Date: Mon, 17 Jan 2022 19:58:53 +0100
Subject: [PATCH] Corona::containmentForScreen: Ignore activities when an empty
 string is used

With the other overloads being deprecated, consumers are forced to put in an empty string
for the activity. As one can see in the pre-existing code, an empty string indicates
that the activity should be ignored. Consequently we should skip the activity check, when
we got an empty string.

BUG: 448590
---
 src/plasma/corona.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plasma/corona.cpp b/src/plasma/corona.cpp
index f034d7b2e..8f532d690 100644
--- a/src/plasma/corona.cpp
+++ b/src/plasma/corona.cpp
@@ -206,7 +206,7 @@ Containment *Corona::containmentForScreen(int screen, const QString &activity, c
     for (Containment *cont : std::as_const(d->containments)) {
         /* clang-format off */
         if (cont->lastScreen() == screen
-            && (cont->activity().isEmpty() || cont->activity() == activity)
+            && ((cont->activity().isEmpty() || activity.isEmpty()) || cont->activity() == activity)
             && (cont->containmentType() == Plasma::Types::DesktopContainment
                 || cont->containmentType() == Plasma::Types::CustomContainment)) { /* clang-format on */
             containment = cont;
-- 
GitLab

